python/lsst/pipe/drivers/coaddDriver.py

```diff
@@ -268,12 +272,16 @@ def coadd(self, cache, data):
         if coadd is None:
             return
 
-        with self.logOperation("detection on %s" % (patchRef.dataId,), catch=True):
-            idFactory = self.detectCoaddSources.makeIdFactory(patchRef)
-            # This includes background subtraction, so do it before writing the
-            # coadd
-            detResults = self.detectCoaddSources.runDetection(coadd, idFactory)
-            self.detectCoaddSources.write(coadd, detResults, patchRef)
+        if self.config.doDetection:
+            with self.logOperation("detection on {}".format(patchRef.dataId),
+                                   catch=True):
+                idFactory = self.detectCoaddSources.makeIdFactory(patchRef)
+                # This includes background subtraction, so do it before writing
+                # the coadd
+                detResults = self.detectCoaddSources.runDetection(coadd, idFactory)
+                self.detectCoaddSources.write(coadd, detResults, patchRef)
+        else:
+            self.assembleCoadd.writeCoaddOutput(patchRef, coadd)
```

Comment:
Would you please add a note or comment noting this subtle point: if detection hasn't been done it writes to `deepCoadd`, but if it has been done then it writes to `deepCoadd_calexp`.